### PR TITLE
feat: add CloudSqlProxyTrait

### PR DESCRIPTION
**Pull request by bshaffer**

### CODE REVIEW

File: test/TestUtils/CloudSqlProxyTraitTest.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    /**
+     * @expectedException Exception
+     * @expectedExceptionMessage Failed to start cloud_sql_proxy
+     * @runInSeparateProcess
+     */
+    public function testFailedRunThrowsException()
+    {
+        $connectionString = 'invalid';
+        $socketDir = '/tmp/cloudsql';
```
Comment (saranshdhingra) [inline test/TestUtils/CloudSqlProxyTraitTest.php:65]: Not the `DB_SOCKET_DIR`?

Comment (saranshdhingra) [reply test/TestUtils/CloudSqlProxyTraitTest.php:65]: Oh, it's an invalid connection string, so it doesn't matter! 👍

File: test/bin/cloud_sql_proxy [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (saranshdhingra) [inline test/bin/cloud_sql_proxy:1]: Not clear on the reasoning behind this file.

**Review by saranshdhingra — APPROVED** [review]
LGTM, just need to understand why we added the `cloud_sql_proxy` file.

File: test/bin/cloud_sql_proxy (hunk 2 above)
Comment (bshaffer) [reply test/bin/cloud_sql_proxy:1]: It's a good question. Mostly i struggled to set up the actual proxy on GitHub actions for over an hour and finally just gave up and wrote a fake one in five minutes. I can add a comment though!

**Pull request merged**